### PR TITLE
Add status icons on the status page

### PR DESCRIPTION
**Pull request by Julow**
So it's easier to see which job failed.

### CODE REVIEW

File: api/client.ml [hunk 1]
```diff
@@ -6,21 +6,26 @@ type variant = string
 
 module Ref_map = Map.Make(String)
 
+module State = struct
+  open Raw.Reader.JobInfo.State
+
+  type t = Raw.Reader.JobInfo.State.unnamed_union_t
```
Comment (CraigFe) [inline api/client.ml]:
```suggestion
  type t = unnamed_union_t
```

File: web-ui/github.ml [hunk 2]
```diff
@@ -36,6 +36,26 @@ let github_branch_url ~owner ~name ref =
...
+  .statuses > li:before {
+    display: inline-block;
+    width: 1em;
+    margin-right: 0.5em;
+    margin-left: -1.5em;
+    text-align: center;
+    line-height: 1.1em;
+  }
+  .statuses > li.not-started:before { content: "●"; color:grey; }
+  .statuses > li.aborted:before { content: "●"; color:red; }
```
Comment (CraigFe) [inline web-ui/github.ml]:
```suggestion
  .statuses > li.aborted:before { content: "A"; color:red; }
```
I still prefer 'A' here, as is being done in Citty. A red dot doesn't look like 'aborted' to me, and this symbol is already being shared.

File: web-ui/github.ml [hunk 3]
```diff
@@ -49,6 +69,22 @@ let breadcrumbs steps page_title =
     List.rev steps
   )
 
+let statuses ss =
+  let open Tyxml.Html in
+  let render_status (s, elms) =
+    let s_class =
```
Comment (CraigFe) [inline web-ui/github.ml]:
```suggestion
    let status_class_name =
```
I don't like the similarity with `a_class` which is something completely different.

**Comment by talex5** [comment]
Having status icons here is really nice. It would be even more useful if we could do something different for `[SKIP]` results.

File: web-ui/github.ml (hunk 3 above)
Comment (Julow) [reply web-ui/github.ml]: Good idea

**Comment by Julow** [comment]
Having a different status for `[SKIP]` would be great ! I'll try to add it

**Comment by Julow** [comment]
Note that I haven't been able to test this PR yet

**Comment by CraigFe** [comment]
Great, thanks!

**Pull request merged**